src/input.c

```diff
@@ -605,6 +733,20 @@ static void parse_char(input_t *restrict input, char ch)
                     input->da_count = input->parm_count;
                 }
                 device_attributes_report(input);
+
+            } else if (ch == 'u') {
+                memset(&input->csi_u_params, 0, sizeof(input->csi_u_params));
+                input->csi_u_params.keycode = input->parms[0];
+                set_csi_u_params(input);
+
+                if (input->csi_u_params.keycode == 99 &&
+                    input->csi_u_params.modifiers == 5) {
+                    /* Ctrl+C - immediate exit */
+                    atomic_store_explicit(&input->exit_requested, true,
+                                          memory_order_relaxed);
+                    return;
+                }
+                ascii_key(input, ch);
```

Comment:
Inside `ascii_key()`, when KKP is active, the ch parameter is completely ignored - csi_u_params.keycode is used instead. This works but violates function contract semantics.

Fix: Extract KKP key dispatch to separate function:
```c
  static void dispatch_kkp_key(input_t *restrict input);
```